### PR TITLE
libobs/UI: Allow launch of custom "properties" window

### PR DESCRIPTION
**Pull request by Andersama**
Makes use of obs_register_modeless_ui to create custom Qt elements

Currently assumes a target of "qt" or "qt dialog" and that id
matches the source id of the source to customize, and that the
task is "properties".

A create callback is used and is passed the source and parent
widget as parameters.

image
image

### CODE REVIEW

**Comment by SuslikV** [comment]
So, properties window may be closed?

**Comment by Andersama** [comment]
@SuslikV the close button for dockable windows not setup in a dockable area is to collapse back down into it's original position. Just imagine that you can alternate between the two screenshots at will.

**Comment by SuslikV** [comment]
This differ from the main UI behavior. It's no good.

**Comment by SuslikV** [comment]
Maybe all elements should be dockable and nested?

File: UI/window-basic-filters.cpp [hunk 1]
```diff
@@ -190,17 +197,41 @@ void OBSBasicFilters::UpdatePropertiesView(int row, bool async)
...
-	view->setMaximumHeight(250);
-	view->setMinimumHeight(150);
-	ui->rightLayout->addWidget(view);
-	view->show();
+	QVBoxLayout *layout = new QVBoxLayout();
+	QDockWidget *dock = new QDockWidget(this);
+	dock->setWindowTitle("Properties");
+	customView = dock;
+	customView->setLayout(layout);
+	auto floatingChanged = [=](const bool _floating) {
```
Comment (DDRBoxman) [inline UI/window-basic-filters.cpp]: Pop out code shouldn't be part of this PR.
